src/main/java/hudson/plugins/git/browser/GitBlitRepositoryBrowser.java

```diff
@@ -105,7 +103,7 @@ public FormValidation doCheckRepoUrl(@AncestorInPath Item project, @QueryParamet
                 return FormValidation.ok();
             }
             FormValidation response;
-            if (checkURIFormat(cleanUrl)) {
+            if (checkURIFormat(cleanUrl, "gitblit")) {
```

Comment:
I don't think you can reasonably assume that a gitblit server must include gitblit in the hostname.  Assembla only provides a hosted solution, so it is safe to check their specific hostname.  [Gitblit](http://gitblit.com/) provides a solution that is either hosted or on premise.

Reply:
Okay, I never anticipated this obvious case. My implementation was based on this assumption that the hostname of the URL will contain the browser name (for ex: http://gitblit.example.com/).

As far as my understanding goes, either I can: 
- remove this assumption, but then would have to provide a custom implementation for Assembla, i.e making checkURIFormat an abstract method.
- branch the logic into two cases, hosted check and on-premise check. 

What do you think?

Reply:
I think that Assembla is the exception in this case.  Most git hosting providers allow a hosted implementation and an implementation that is installed and maintained inside the user network.

I think it would be better to place checkURIFormat inside the Assembla specific class and then place something more general in the GitRepositoryBrowser class.

Reply:
> Most git hosting providers allow a hosted implementation and an implementation that is installed and maintained inside the user network.
This means that creating a simple host name check is not feasible, hence I have created a separate `checkURIFormatAndHostName` for AssemblaWeb since Assembla provides a hosted solution only. For others, it's just a plain UrlValidatior check. For reference, the commit is 454397e .